packages/api-fetch/src/middlewares/preloading.js

```diff
@@ -45,7 +48,12 @@ function createPreloadingMiddleware( preloadedData ) {
 			const method = options.method || 'GET';
 			const path = getStablePath( options.path );
 
-			if ( 'GET' === method && cache[ path ] ) {
+			if (
+				'GET' === method &&
+				cache[ path ] &&
+				! cache[ path ].hasBeenPreloaded
+			) {
+				cache[ path ].hasBeenPreloaded = true;
```

Comment:
maybe instead of adding a boolean we can just unset the cache key?

Reply:
That's a great point! Unsetting would probably work as well and be less verbose. 👍